infra/modules/azure_storage_account/cmk.tf

```diff
@@ -0,0 +1,38 @@
+# tfsec:ignore:azure-keyvault-ensure-key-expiry
+resource "azurerm_key_vault_key" "key" {
+  for_each     = (local.cmk_flags.kv && var.customer_managed_key.key_name == null ? toset(["kv"]) : toset([]))
+  name         = "${replace("${module.naming_convention.prefix}-st-${module.naming_convention.suffix}", "-", "")}-cmk-kv"
```

Comment:
What do you think of a format like `io-p-itn-<domain>-[<app_name>]-st-cmk-01`? I took it from the current PEP format as `io-p-itn-wallet-sql-pep-01`